<launch>
  <include file="$(find pr2_machine)/$(env ROBOT).machine" />

  <!-- set tilt laser to look straight ahead -->
  <node pkg="pr2_mechanism_controllers" type="send_periodic_cmd_srv.py" name="laser_tilt_controller_3dnav_params" args="laser_tilt_controller linear 1.0 0 0" />


  <!-- map -->
  <!--<node name="map_server" pkg="map_server" type="map_server" args="$(find sbpl_loop_env_global_planner)/maps/newell_simon_rotated_5cm.pgm 0.05" respawn="false" />-->
  <!--<node name="map_server" pkg="map_server" type="map_server" args="$(find sbpl_loop_env_global_planner)/map_test.pgm 0.05" respawn="false" />-->
  <node name="map_server" pkg="map_server" type="map_server" args="$(find sbpl_loop_env_global_planner)/sbpl_loop2.pgm 0.05" respawn="false" />

  <!-- localization -->
  <node pkg="amcl" type="amcl" name="amcl" machine="c2">
    <remap from="scan" to="base_scan" />
    <!-- Publish scans from best pose at a max of 10 Hz -->
    <param name="base_frame_id" value="base_footprint"/>
    <param name="odom_model_type" value="omni"/>
    <param name="odom_alpha5" value="0.1"/>
    <param name="transform_tolerance" value="0.2" />
    <param name="gui_publish_rate" value="10.0"/>
    <param name="laser_max_beams" value="30"/>
    <param name="min_particles" value="500"/>
    <param name="max_particles" value="5000"/>
    <param name="kld_err" value="0.05"/>
    <param name="kld_z" value="0.99"/>
    <param name="odom_alpha1" value="0.2"/>
    <param name="odom_alpha2" value="0.2"/>
    <!-- translation std dev, m -->
    <param name="odom_alpha3" value="0.8"/>
    <param name="odom_alpha4" value="0.2"/>
    <param name="laser_z_hit" value="0.5"/>
    <param name="laser_z_short" value="0.05"/>
    <param name="laser_z_max" value="0.05"/>
    <param name="laser_z_rand" value="0.5"/>
    <param name="laser_sigma_hit" value="0.2"/>
    <param name="laser_lambda_short" value="0.1"/>
    <param name="laser_lambda_short" value="0.1"/>
    <param name="laser_model_type" value="likelihood_field"/>
    <!-- <param name="laser_model_type" value="beam"/> -->
    <param name="laser_likelihood_max_dist" value="2.0"/>
    <param name="update_min_d" value="0.2"/>
    <param name="update_min_a" value="0.5"/>
    <param name="odom_frame_id" value="odom_combined"/>
    <param name="resample_interval" value="1"/>
    <param name="transform_tolerance" value="0.1"/>
    <param name="recovery_alpha_slow" value="0.0"/>
    <param name="recovery_alpha_fast" value="0.0"/>
  </node>

  <!-- move base with sbpl_dynamic_planner -->
  <include file="$(find sbpl_loop_env_global_planner)/config/move_base_sbpl_pr2.xml"/>

  <!-- PR2 head behavior node - the head looks at the ROI closest to the robot -->
  <node name="pr2_head_behavior" pkg="pr2_head_behavior" type="pr2_head_behavior" respawn="true" />

  <!-- Launch the kinect -->
  <!-- <include file="$(find kalman_filter)/launch/openni-electric.launch"/>-->

  <!-- Bring up the octomap-server to integrate kinect point clouds -->
  <node name="octomap_server" pkg="octomap_server" type="octomap_server_node">
    <!--<remap from="cloud_in" to="openni/depth_registered/points"/>-->
    <remap from="cloud_in" to="narrow_stereo/points2"/>
    <remap from="map" to="projected_map"/>
  </node>

</launch>
